RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -291,32 +634,695 @@ class DeepTauId : public deep_tau::DeepTauBase {
+            else
+                throw cms::Exception("DeepTauId") << "version " << version << " is not supported.";
+            for(int k = 0; k < dnn_inputs_2017v1::NumberOfOutputs; ++k) {
+                const float pred = pred_vector[0].flat<float>()(k);
+                if(!(pred >= 0 && pred <= 1))
+                    throw cms::Exception("DeepTauId") << "invalid prediction = " << pred << " for tau_index = "
+                                                      << tau_index << ", pred_index = " << k;
+                predictions.matrix<float>()(tau_index, k) = pred;
+            }
         }
         return predictions;
     }
 
+    void getPredictionsV1(const TauType& tau, const pat::ElectronCollection& electrons,
+                          const pat::MuonCollection& muons, std::vector<tensorflow::Tensor>& pred_vector)
+    {
+        const tensorflow::Tensor& inputs = createInputsV1<dnn_inputs_2017v1>(tau, electrons, muons);
+        tensorflow::run(&(cache_->getSession()), { { input_layer_, inputs } }, { output_layer_ }, &pred_vector);
+    }
+
+    void getPredictionsV2(const TauType& tau, const pat::ElectronCollection& electrons,
+                          const pat::MuonCollection& muons, const pat::PackedCandidateCollection& pfCands,
+                          const reco::Vertex& pv, double rho, std::vector<tensorflow::Tensor>& pred_vector)
+    {
+        CellGrid inner_grid(11, 11, 0.02, 0.02);
+        CellGrid outer_grid(21, 21, 0.05, 0.05);
+        fillGrids(tau, electrons, inner_grid, outer_grid);
+        fillGrids(tau, muons, inner_grid, outer_grid);
+        fillGrids(tau, pfCands, inner_grid, outer_grid);
+
+        createTauBlockInputs(tau, pv, rho);
+        createEgammaBlockInputs(tau, pv, rho, electrons, pfCands, inner_grid, true);
+        createMuonBlockInputs(tau, pv, rho, muons, pfCands, inner_grid, true);
+        createHadronsBlockInputs(tau, pv, rho, pfCands, inner_grid, true);
+        createEgammaBlockInputs(tau, pv, rho, electrons, pfCands, outer_grid, false);
+        createMuonBlockInputs(tau, pv, rho, muons, pfCands, outer_grid, false);
+        createHadronsBlockInputs(tau, pv, rho, pfCands, outer_grid, false);
+
+        tensorflow::run(&(cache_->getSession()),
+            { { "input_tau", *tauBlockTensor_ },
+              { "input_inner_egamma", *eGammaInnerTensor_}, { "input_outer_egamma", *eGammaOuterTensor_ },
+              { "input_inner_muon", *muonInnerTensor_ }, { "input_outer_muon", *muonOuterTensor_ },
+              { "input_inner_hadrons", *hadronsInnerTensor_ }, { "input_outer_hadrons", *hadronsOuterTensor_ } },
+            { "main_output/Softmax" }, &pred_vector);
+
+    }
+
+    template<typename Collection>
+    void fillGrids(const TauType& tau, const Collection& objects, CellGrid& inner_grid, CellGrid& outer_grid)
+    {
+        static constexpr double outer_dR2 = std::pow(0.5, 2);
```

Comment:
Do not initialize `static constexpr` with a function but by a number. Fixed in the 10_6_X version by commit 8fcee1d3599daa2d0ba310dcb782f986d0210237